### ISSUE TITLE
Flow is working but flow error visual indicators are not going away once error is resolved

### ISSUE THREAD

**Issue by wagwes** [w0rp/ale#1884]
This plugin is so freaking awesome, thank you so much for all your hard work on it. I tried looking to see if anyone else was running into this specific bug but google and searching in issues for the repo hasn't yielded anything fruitful so here we go:

## Information

**VIM version**
```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Jul 22 2018 05:24:30)
macOS version
```

Operating System: OSX High Sierry: 10.13.6

### :ALEInfo
```

 Current Filetype: javascript.jsx
Available Linters: ['eslint', 'flow', 'flow-language-server', 'jscs', 'jshint', 'standard', 'tsserver', 'xo']
  Enabled Linters: ['eslint', 'flow', 'flow-language-server', 'jscs', 'jshint', 'standard', 'tsserver', 'xo']
 Suggested Fixers: 
  'eslint' - Apply eslint --fix to a file.
  'importjs' - automatic imports for javascript
  'prettier' - Apply prettier to a file.
  'prettier_eslint', 'prettier-eslint' - Apply prettier-eslint to a file.
  'prettier_standard', 'prettier-standard' - Apply prettier-standard to a file.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'standard' - Fix JavaScript files using standard --fix
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'xo' - Fix JavaScript files using xo --fix.
 Linter Variables:

let g:ale_javascript_eslint_executable = 'eslint'
let g:ale_javascript_eslint_options = ''
let g:ale_javascript_eslint_suppress_eslintignore = 0
let g:ale_javascript_eslint_suppress_missing_config = 0
let g:ale_javascript_eslint_use_global = 0
let g:ale_javascript_flow_executable = 'flow'
let g:ale_javascript_flow_ls_executable = 'flow'
let g:ale_javascript_flow_ls_use_global = 0
let g:ale_javascript_flow_use_global = 0
let g:ale_javascript_flow_use_home_config = 0
let g:ale_javascript_flow_use_respect_pragma = 1
let g:ale_javascript_jscs_executable = 'jscs'
let g:ale_javascript_jscs_use_global = 0
let g:ale_javascript_jshint_executable = 'jshint'
let g:ale_javascript_jshint_use_global = 0
let g:ale_javascript_standard_executable = 'standard'
let g:ale_javascript_standard_options = ''
let g:ale_javascript_standard_use_global = 0
let g:ale_javascript_tsserver_config_path = ''
let g:ale_javascript_tsserver_executable = 'tsserver'
let g:ale_javascript_tsserver_use_global = 0
let g:ale_javascript_xo_executable = 'xo'
let g:ale_javascript_xo_options = ''
let g:ale_javascript_xo_use_global = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'always'
let g:ale_lint_on_insert_leave = 0
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_list_window_size = 10
let g:ale_list_vertical = 0
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = 'X'
let g:ale_sign_info = '?'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = 'X'
let g:ale_sign_style_warning = '?'
let g:ale_sign_warning = '?'
let g:ale_statusline_format = ['X %d', '? %d', '']
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/.bin/flow'' --version; echo']
(executable check - failure) jscs
(executable check - failure) jshint
(executable check - failure) standard
(executable check - failure) xo
(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/.bin/flow'' check-contents --respect-pragma --json --from ale ''/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js'' < ''/var/folders/d6/_9k9s5xn60l87p59rcyyzy5h0000gp/T/v38rSuH/17157/ButtonWithSpinner.js''; echo']

<<<OUTPUT STARTS>>>
{"flowVersion":"0.72.0","jsonVersion":"1","errors":[{"kind":"infer","level":"error","suppressions":[],"extra":[{"message":[{"context":null,"descr":"References:","type":"Blame","path":"","line":0,"endline":0,"start":1,"end":0}]},{"message":[{"context":"  buttonStyle?: 'light' | 'secondary' | 'primary',","descr":"[1]","type":"Blame","loc":{"source":"/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js","type":"SourceFile","start":{"line":11,"column":17,"offset":208},"end":{"line":11,"column":49,"offset":241}},"path":"/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js","line":11,"endline":11,"start":17,"end":49}]}],"message":[{"context":"      className={`btn btn-${buttonStyle} btn-block`}","descr":"Cannot coerce `buttonStyle` to string because undefined [1] should not be coerced.","type":"Blame","loc":{"source":"/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js","type":"SourceFile","start":{"line":17,"column":29,"offset":418},"end":{"line":17,"column":39,"offset":429}},"path":"/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js","line":17,"endline":17,"start":29,"end":39}]}],"passed":false}
<<<OUTPUT ENDS>>>

(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/eslint/bin/eslint.js'' -f unix --stdin --stdin-filename ''/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js'' < ''/var/folders/d6/_9k9s5xn60l87p59rcyyzy5h0000gp/T/v38rSuH/17158/ButtonWithSpinner.js''']

<<<NO OUTPUT RETURNED>>>

(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/.bin/flow'' --version; echo']
(executable check - failure) jscs
(executable check - failure) jshint
(executable check - failure) standard
(executable check - failure) xo
(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/.bin/flow'' check-contents --respect-pragma --json --from ale ''/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js'' < ''/var/folders/d6/_9k9s5xn60l87p59rcyyzy5h0000gp/T/v38rSuH/17159/ButtonWithSpinner.js''; echo']

<<<OUTPUT STARTS>>>
{"flowVersion":"0.72.0","jsonVersion":"1","errors":[],"passed":true}
<<<OUTPUT ENDS>>>

(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/eslint/bin/eslint.js'' -f unix --stdin --stdin-filename ''/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js'' < ''/var/folders/d6/_9k9s5xn60l87p59rcyyzy5h0000gp/T/v38rSuH/17160/ButtonWithSpinner.js''']

<<<NO OUTPUT RETURNED>>>

(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/.bin/flow'' --version; echo']
(executable check - failure) jscs
(executable check - failure) jshint
(executable check - failure) standard
(executable check - failure) xo
(finished - exit code 0) ['/bin/zsh', '-c', '''/Users/weschiang/Code/wagwebapp/node_modules/.bin/flow'' check-contents --respect-pragma --json --from ale ''/Users/weschiang/Code/wagwebapp/src/components/ButtonWithSpinner/ButtonWithSpinner.js'' < ''/var/folders/d6/_9k9s5xn60l87p59rcyyzy5h0000gp/T/v38rSuH/17161/ButtonWithSpinner.js''; echo']

<<<OUTPUT STARTS>>>
{"flowVersion":"0.72.0","jsonVersion":"1","errors":[],"passed":true}
<<<OUTPUT ENDS>>>
```

## What went wrong
ALE is linting flow types correctly but after I resolve the error, it would appear that the error is not going away. If I close the file and then re-open it, the error is now gone.

This only happens for flow types and not others, i.e. eslint.

## Reproducing the bug
1. Create a React component with flow typing enabled:
```
// @flow
import React from 'react';

type Props = {
  isFetching?: boolean,
  disabled?: boolean,
  type?: string,
  value?: string,
  [string]: any,
  buttonStyle: 'light' | 'secondary' | 'primary',
};

const ButtonWithSpinner = ({ isFetching, disabled, type, buttonStyle, value, ...rest }: Props) => (
  <div className="form-group">
    <button
      className={`btn btn-${buttonStyle} btn-block`}
      type={type}
      disabled={disabled}
      style={isFetching ? { padding: '0', height: '50px' } : {}}
      {...rest}
    >
      {isFetching ? <div>spinner</div> : value}
    </button>
  </div>
);

ButtonWithSpinner.defaultProps = {
  isFetching: false,
  disabled: false,
  type: 'submit',
  value: '',
  buttonStyle: 'light',
};

export default ButtonWithSpinner;
```
2. Edit the file such that a flow error is detected.
3. Fix the flow error
4. Observe that the error does not go away after resolving the flow error

**Comment by wagwes** [comment]
UPDATE: I updated my ALE plugin to the latest release but am still experiencing the same issue. However, I've discovered that instead of closing and reopening the file, I can just type `:ALEReset`. Though this is faster than before, it would still be really nice to have flow linting work just like other linters i.e. ESlint

**Comment by redbmk** [comment]
I use `let g:ale_echo_msg_format = '[%linter%] %s [%severity%% code%]'` so that I can see what is causing the errors. I've noticed I sometimes get errors from `flow` and sometimes from `flow-language-server`. Whenever this error occurs it's always `flow-language-server` that's throwing the left-over error.

I first noticed this after updating `w0rp/ale` a while ago...trying to figure out what version broke it for me. Could be a coincidence though - maybe I updated something else at the same time
